### ISSUE TITLE
Unexplicit error message when initialising global variables

### ISSUE THREAD

**Issue by lforg37** [serge-sans-paille/pythran#586]
The following code produced an unexplicit error message when I tried to "pythranise" it : 
```
import numpy as np

xmax = 1000001
ymax = 12000
table = np.zeros(xmax)
table[1] = 1

#pythran export syracuse(int)
def syracuse(x):
    if x < ymax and table[x] != 0:
        return table[x]

    suivant = 0
    if x % 2 == 0:
        suivant = x/2
    else:
        suivant = 3*x + 1
    retour = 1 + syracuse(suivant)
    if x < ymax:
        table[x] = retour
    return retour

#pythran export allsyracuse()
def allsyracuse():
    lmax = 0
    nmax = 0
    for i in range(1, xmax):
        taille = syracuse(i)
        if taille > lmax:
            lmax = taille
            nmax = i

    return (nmax, lmax)

```

The error message is 
```
Traceback (most recent call last):
  File "/home/luc/workspace/formationPytran/venv/bin/pythran", line 11, in <module>
    sys.exit(run())
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/run.py", line 133, in run
    **compile_flags(args))
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/toolchain.py", line 346, in compile_pythranfile
    **kwargs)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/toolchain.py", line 303, in compile_pythrancode
    module = generate_cxx(module_name, pythrancode, specs, opts)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/toolchain.py", line 132, in generate_cxx
    refine(pm, ir, optimizations)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/middlend.py", line 18, in refine
    pm.apply(ExpandGlobals, node)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/passmanager.py", line 215, in apply
    res = a.apply(node, ctx)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/passmanager.py", line 177, in apply
    new_node = self.run(node, ctx)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/passmanager.py", line 171, in run
    n = super(Transformation, self).run(node, ctx)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/passmanager.py", line 114, in run
    return self.visit(node)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/passmanager.py", line 80, in visit
    return super(ContextManager, self).visit(node)
  File "/usr/lib64/python2.7/ast.py", line 241, in visit
    return visitor(node)
  File "/home/luc/workspace/formationPytran/venv/lib/python2.7/site-packages/pythran/transformations/expand_globals.py", line 50, in visit_Module
    assert isinstance(target, ast.Name)
AssertionError
```

Moving the global initialisation inside the allsyracuse function solved the problem.

**Comment by pbrunet** [comment]
Reason is : We don't support globals variables. indeed, we accept globals only in Read-Only while you assign an element which is not allowed. But I agree that error message could be improved.

**Comment by serge-sans-paille** [comment]
@pbrunet: I got the fix (better error reporting)

**Comment by serge-sans-paille** [comment]
Fixed by 1ca2cd2fc77aa328216658201c536b378c1d997a
